src/SlnGen.Build.Tasks/SlnGen.cs

```diff
@@ -123,17 +123,25 @@ internal IEnumerable<string> GetSolutionItems()
         /// </summary>
         /// <param name="fileExists">A <see cref="Func{String, Boolean}"/> to use when determining if a file exists.</param>
         /// <returns>An <see cref="IEnumerable{String}"/> of full paths to include as solution items.</returns>
-        internal IEnumerable<string> GetSolutionItems(Func<string, bool> fileExists)
+        internal IEnumerable<SlnItem> GetSolutionItems(Func<string, bool> fileExists)
         {
-            foreach (string solutionItem in SolutionItems.Select(i => i.GetMetadata("FullPath")).Where(i => !String.IsNullOrWhiteSpace(i)))
+            foreach (var solutionItem in SolutionItems)
             {
-                if (!fileExists(solutionItem))
+                var fullPath = solutionItem.GetMetadata("FullPath");
+
+                if (String.IsNullOrWhiteSpace(fullPath))
```

Comment:
Do you know under what conditions `fullPath` would be null?  Just curious...

Reply:
An incorrect `IItemTask` implementation? Anyway, it won't be null under normal conditions, but I wanted to check for white space and this is the easiest way I think?